src/sagas/RootSaga.js

```diff
@@ -38,29 +38,39 @@ import { keepKeys } from '../services/utils';
 
 export function* handleFetchProfile(action) {
   const { username } = action.payload;
-  const currentUsername = yield select(state => state.authentication.username); // eslint-disable-line
+  const { isCurrentUserProfile, userAccount } = yield select(handleFetchProfileSelector);
+
+  // Default our data.
+  let preferences = {};
+  let account = userAccount;
```

Comment:
Assume we'll use the current user account data for the page - this is overridden below if we're not looking at the current user's profile.